ihmc-java-toolkit/src/main/java/us/ihmc/tools/thread/LoopingThread.java

```diff
@@ -0,0 +1,193 @@
+package us.ihmc.tools.thread;
+
+import us.ihmc.commons.Conversions;
+import us.ihmc.commons.RunnableThatThrows;
+import us.ihmc.commons.exception.ExceptionHandler;
+import us.ihmc.commons.exception.ExceptionTools;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * A thread that executes code in a loop.
+ * Optionally, you may set a limit to the loop frequency.
+ * This thread does not finish running until {@link #close()} or {@link #blockingClose()} is called.
+ * Once started, make sure to close this thread.
+ */
+public class LoopingThread extends Thread
```

Comment:
What you're going for seems a lot like what I had done with PausablePeriodicThread. That name seems like it would be a better choice than LoopingThread.

Reply:
If you want to remove PausablePeriodicThread and replace it with this, I would be in favor. I don't like that that one uses the ExceptionHandlingThreadScheduler.

Reply:
Yeah this is pretty similar to PausablePeriodicThread. Should we deprecate that one in favor of this one. I personally don't have a problem with the name LoopingThread.

Reply:
Hmmm I'm not sure if I'd call this "periodic" since it can free spin the loop. I suppose there's still some period that the loop runs in, but it wouldn't be defined by the programmer.

Reply:
Fair. Would PausableLoopingThread be better than LoopingThread?

Reply:
Sure, I'm cool with that